src/hotspot/share/opto/vectornode.cpp

```diff
@@ -1172,7 +1182,8 @@ Node* VectorNode::degenerate_vector_rotate(Node* src, Node* cnt, bool is_rotate_
 Node* RotateLeftVNode::Ideal(PhaseGVN* phase, bool can_reshape) {
   int vlen = length();
   BasicType bt = vect_type()->element_basic_type();
-  if (!Matcher::match_rule_supported_vector(Op_RotateLeftV, vlen, bt)) {
+  int extinfo = encode_rotate_vector_shift_type(in(2));
+  if (!Matcher::match_rule_supported_vector(Op_RotateLeftV, vlen, bt, extinfo)) {
```

Comment:
I don't see much value in encoding and packing node-specific info in an int. Why not simply introduce a new platform-specific entry and pass a `Node*` instead letting platform-specific code to extract any useful information from it?

As an alternative, introduce a new capability (`Matcher::supports_vector_variable_rotates`) and check it in shared code (`RotateLeftVNode::Ideal`).

Reply:
Introduced `Matcher::supports_vector_variable_rotates(VectorNode *n)` in the updated commit https://github.com/openjdk/jdk/pull/1532/commits/d2cb793c35f02667927acfa76a115442b1d6956d.
Verified it both on `AARCH64` and `X86_64` platforms with `fastdebug` build. Tests are good.